### DISCUSSION TITLE
Splitting steps corrupts step_count

### DISCUSSION THREAD

**Discussion by rusty-snake** [openstreetmap/iD#10534] [General]
This is a follow-up on #8069, so ping @matkoniecz @quincylvania @DerDings @westnordost

---

The current behaviour of iD when splitting ways is to create two ways with identical tags. So if you have a 10m footway with `surface=asphalt` but actually the first half is `surface=pacing_stone`, you place a node, split the way and change the `surface`. The same applies for all other kinds like `access*`, `smoothness`, `cycleway`, `tracktype`, ... Except for `step_count` which gets very silently modified. If you split a `highway=steps` to change a steps related tag such as `width`, this is good (i.e. no double count). If you however go to change one of the two ways to a `highway=footway` because there was just a long `highway=steps` mapped where half of the distance is actually only the footway to it, you end up with

1. a `highway=footway` with `step_count=*`. If it has steps, it shouldn't be a footway but if it does not have steps, the tag does not make sense.
2. a `highway=steps` with an incorrect step count. Sometimes even `step_count=0`.

So especially in the case of changing a `highway=steps` to a `highway=footway|path` it would be good to nuke the `step_count`, mark it for re-survey or somehow notify the user that they did not got an identical copy of the tags.

**Comment by matkoniecz** [comment]
> The current behaviour of iD when splitting ways is to create two ways with identical tags. 

per https://github.com/openstreetmap/iD/issues/8069#issuecomment-706176064 and https://github.com/openstreetmap/iD/commit/d43fc62994476027b751cdec08acfaa5c2baa95d step_count is handled differently

has it changed

**Reply by rusty-snake** [reply]
Comes later: 

> Except for `step_count` which gets very silently modified.

It didn't change. The problem is that is happens very silently and can hurt sometimes. As it the only(?) tag that gets modified by the split action it happens unexpected.
